### PR TITLE
feat: split schematics component context

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change
split schematics context of the schematics component

### CODE REVIEW

File: packages/@o3r/core/collection.json [hunk 1]
```diff
@@ -22,6 +22,12 @@
       "factory": "./schematics/component/index#ngGenerateComponent",
       "schema": "./schematics/component/schema.json"
     },
+    "context-to-component": {
```
Comment (kpanot) [inline packages/@o3r/core/collection.json:25]: should be added in readme.md

File: packages/@o3r/core/schematics/index.it.spec.ts [hunk 2]
```diff
@@ -160,6 +161,10 @@ describe('new Otter application', () => {
     execSync('yarn ng g @o3r/analytics:add-analytics --defaults=true --path="src/components/test-add-analytics-component/container/test-add-analytics-component-cont.component.ts"', execAppOptions);
     addImportToAppModule('TestAddAnalyticsComponentContModule', 'src/components/test-add-analytics-component');
 
+    execSync('yarn ng g @o3r/core:component--defaults=true test-add-context-component ${defaultOptions}', execAppOptions);
```
Comment (mrednic-1A) [inline packages/@o3r/core/schematics/index.it.spec.ts:164]: Is it working without a space before --default ?

File: packages/@o3r/core/schematics/component/common/context.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+          }
+        ], 0, null);
+      } else {
+        context.logger.info(`
+          You can add context later to this component via this command:
+          ng g ${contextPackageName}:${addContextSchematicName} --path="${componentPath}"
+        `);
+      }
+    }
+  } catch {
```
Comment (mrednic-1A) [inline packages/@o3r/core/schematics/component/common/context.ts:45]: I think you can remove the try/catch. the script cannot be run if `o3r/core` is not already installed.

File: packages/@o3r/core/schematics/component/common/context.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  componentPath: string,
+  options: Pick<NgGenerateComponentSchematicsSchema, 'useContext' | 'skipLinter'>,
+  context: SchematicContext
+): Promise<Rule[]> => {
+  let useContext = options.useContext;
+  let alwaysUseContext;
+  try {
+    require.resolve(`${contextPackageName}/package.json`);
+    if (useContext === null && context.interactive) {
+      useContext = await askConfirmation('Generate component with Otter context ?', true);
```
Comment (vscaiceanu-1a) [inline packages/@o3r/core/schematics/component/common/context.ts:19]:
```suggestion
      useContext = await askConfirmation('Generate component with Otter context?', true);
```

File: packages/@o3r/core/schematics/index.it.spec.ts (hunk 2 above)
Comment (vscaiceanu-1a) [inline packages/@o3r/core/schematics/index.it.spec.ts:164]:
```suggestion
    execSync('yarn ng g @o3r/core:component --defaults=true test-add-context-component ${defaultOptions}', execAppOptions);
```

File: packages/@ama-sdk/generator-sdk/src/generators/shell/templates/base/package.json [hunk 5]
```diff
@@ -84,7 +84,7 @@
     "@o3r/eslint-plugin": "^<%= sdkCoreVersion %>",
     "@swc/cli": "^0.1.57",
     "@swc/core": "^1.2.249",
-    "@types/jasmine": "~4.3.5",
+    "@types/jasmine": "4.3.5",
```
Comment (kpanot) [inline packages/@ama-sdk/generator-sdk/src/generators/shell/templates/base/package.json:87]: Why not keeping the `~`?

**Review by vscaiceanu-1a — APPROVED** [review]

**Review by cpaulve-1A — APPROVED** [review]

**Pull request merged**